django/db/models/functions/comparison.py

```diff
@@ -14,6 +14,20 @@ def as_sql(self, compiler, connection, **extra_context):
         extra_context['db_type'] = self.output_field.cast_db_type(connection)
         return super().as_sql(compiler, connection, **extra_context)
 
+    def as_sqlite(self, compiler, connection, **extra_context):
+        db_type = self.output_field.db_type(connection)
+        if db_type in {'datetime', 'time'}:
+            # Use strftime as datetime/time don't keep fractional seconds.
+            template = 'strftime(%%s, %(expressions)s)'
+            sql, params = super().as_sql(compiler, connection, template=template, **extra_context)
+            format_string = '%H:%M:%f' if db_type == 'time' else '%Y-%m-%d %H:%M:%f'
+            params.insert(0, format_string)
+            return sql, params
+        elif db_type == 'date':
+            template = 'date(%(expressions)s)'
+            return super().as_sql(compiler, connection, template=template, **extra_context)
+        return self.as_sql(compiler, connection, template=self.template, **extra_context)
```

Comment:
Last nit, you don't need to be passing `self.template` here and `super()` will default to it if it's missing.